From ec4d75140b9715a11e44f74e1c8507a64483ea25 Mon Sep 17 00:00:00 2001
From: Michael Adler <therisen06@gmail.com>
Date: Wed, 21 Apr 2021 15:46:05 +0200
Subject: [PATCH 3/8] Disable focus follows mouse

Signed-off-by: Michael Adler <therisen06@gmail.com>
Signed-off-by: Michael Adler <therisen06@gmail.com>
---
 dwm.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/dwm.c b/dwm.c
index e9a3144..20e234f 100644
--- a/dwm.c
+++ b/dwm.c
@@ -310,7 +310,6 @@ static void (*handler[LASTEvent]) (XEvent *) = {
 	[ConfigureRequest] = configurerequest,
 	[ConfigureNotify] = configurenotify,
 	[DestroyNotify] = destroynotify,
-	[EnterNotify] = enternotify,
 	[Expose] = expose,
 	[FocusIn] = focusin,
 	[KeyPress] = keypress,
-- 
2.31.1

