From 0ddeeb09ced8bbddc0bfe091fae082555ba68aa7 Mon Sep 17 00:00:00 2001
From: Michael Adler <adlerm@mailbox.org>
Date: Mon, 31 Oct 2022 15:28:27 +0100
Subject: [PATCH 15/16] fix build

---
 dwm.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/dwm.c b/dwm.c
index 4a9da5b..9ed5c97 100644
--- a/dwm.c
+++ b/dwm.c
@@ -1895,8 +1895,6 @@ sigterm(int unused)
 void
 spawn(const Arg *arg)
 {
-	if (arg->v == dmenucmd)
-		dmenumon[0] = '0' + selmon->num;
 	if (fork() == 0) {
 		if (dpy)
 			close(ConnectionNumber(dpy));
-- 
2.38.1

