From affcacb30eb25adc50066e784a85a028b86758b7 Mon Sep 17 00:00:00 2001
From: Michael Adler <therisen06@gmail.com>
Date: Fri, 1 Oct 2021 08:44:05 +0200
Subject: [PATCH] Do not use venv

---
 coq/__main__.py | 74 +++----------------------------------------------
 1 file changed, 4 insertions(+), 70 deletions(-)

diff --git a/coq/__main__.py b/coq/__main__.py
index 07ce63ed..9720e089 100644
--- a/coq/__main__.py
+++ b/coq/__main__.py
@@ -66,63 +66,6 @@ _IN_VENV = _RT_PY == _EXEC_PATH
 
 
 if command == "deps":
-    assert not _IN_VENV
-
-    io_out = StringIO()
-    try:
-        from venv import EnvBuilder
-
-        print("...", flush=True)
-        with redirect_stdout(io_out), redirect_stderr(io_out):
-            EnvBuilder(
-                system_site_packages=False,
-                with_pip=True,
-                upgrade=True,
-                symlinks=not IS_WIN,
-                clear=True,
-            ).create(_RT_DIR)
-    except (ImportError, CalledProcessError):
-        msg = "Please install python3-venv separately. (apt, yum, apk, etc)"
-        io_out.seek(0)
-        print(msg, io_out.read(), file=stderr)
-        exit(1)
-    else:
-        proc = run(
-            (
-                _RT_PY,
-                "-m",
-                "pip",
-                "install",
-                "--upgrade",
-                "pip",
-            ),
-            cwd=TOP_LEVEL,
-            stdin=DEVNULL,
-            stderr=STDOUT,
-        )
-        if proc.returncode:
-            print("Installation failed, check :message", file=stderr)
-            exit(proc.returncode)
-        proc = run(
-            (
-                _RT_PY,
-                "-m",
-                "pip",
-                "install",
-                "--upgrade",
-                "--force-reinstall",
-                "--requirement",
-                str(REQUIREMENTS),
-            ),
-            cwd=TOP_LEVEL,
-            stdin=DEVNULL,
-            stderr=STDOUT,
-        )
-        if proc.returncode:
-            print("Installation failed, check :message", file=stderr)
-            exit(proc.returncode)
-        else:
-            _LOCK_FILE.write_text(_REQ)
             msg = """
             ---
             You can now use :COQnow
@@ -131,19 +74,10 @@ if command == "deps":
 
 elif command == "run":
     try:
-        lock = _LOCK_FILE.read_text()
-    except Exception:
-        lock = ""
-    try:
-        if not _IN_VENV:
-            raise ImportError()
-        elif lock != _REQ:
-            raise ImportError()
-        else:
-            import pynvim
-            import pynvim_pp
-            import std2
-            import yaml
+        import pynvim
+        import pynvim_pp
+        import std2
+        import yaml
     except ImportError:
         msg = f"""
         Please update dependencies using :COQdeps
-- 
2.33.0

