From 3d77078123f8bb98a315bec30643632091be58a1 Mon Sep 17 00:00:00 2001
From: Michael Adler <therisen06@gmail.com>
Date: Thu, 11 Nov 2021 16:39:46 +0100
Subject: [PATCH] unbind alt-enter

---
 tilish.tmux | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/tilish.tmux b/tilish.tmux
index bc5cf0e..d7ff653 100755
--- a/tilish.tmux
+++ b/tilish.tmux
@@ -186,19 +186,6 @@ else
 	tmux $bind "${mod}${L}" run-shell 'old=`tmux display -p "#{pane_index}"`; tmux select-pane -R; tmux swap-pane -t $old'
 fi
 
-# Open a terminal with Alt + Enter.
-if [ -z "$legacy" ]
-then
-	tmux $bind "${mod}enter" \
-		run-shell 'cwd="`tmux display -p \"#{pane_current_path}\"`"; tmux select-pane -t "bottom-right"; tmux split-pane -c "$cwd"'
-else
-	tmux $bind "${mod}enter" \
-		select-pane -t 'bottom-right' \\\;\
-		split-window \\\;\
-		run-shell 'tmux select-layout' \\\;\
-		send escape
-fi
-
 # Name a window with Alt + n.
 tmux $bind "${mod}n" \
 	command-prompt -p 'Workspace name:' 'rename-window "%%"'
-- 
2.33.1

