From 07f28f4c4920c7ca40c23a5e7efc0d81ae039772 Mon Sep 17 00:00:00 2001
From: Michael Adler <adlerm@mailbox.org>
Date: Tue, 27 Dec 2022 21:12:41 +0100
Subject: [PATCH] fix: support for block devices

---
 window/manager.go | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/window/manager.go b/window/manager.go
index 4e3866c..965bcc6 100644
--- a/window/manager.go
+++ b/window/manager.go
@@ -553,6 +553,24 @@ func (m *Manager) writeFile(r *event.Range, name string) (string, int64, error)
 		window.name = filepath.Base(name)
 		window.mu.Unlock()
 	}
+	if fi, err := os.Lstat(name); err == nil && fi.Mode()&os.ModeType == os.ModeDevice {
+		f, err := os.OpenFile(name, os.O_RDWR, 0)
+		if err != nil {
+			return name, 0, err
+		}
+		defer f.Close()
+		if err := f.Truncate(0); err != nil {
+			return name, 0, err
+		}
+		n, err := window.writeTo(r, f)
+		if err != nil {
+			return name, 0, err
+		}
+		if window.filename == name {
+			window.savedChangedTick = window.changedTick
+		}
+		return name, n, nil
+	}
 	tmpf, err := os.OpenFile(
 		name+"-"+strconv.FormatUint(rand.Uint64(), 16),
 		os.O_RDWR|os.O_CREATE|os.O_EXCL, m.filePerm(name),
-- 
2.39.0

