From 0f9f12da9bee6b4964fed6d87fea678109ee3c3b Mon Sep 17 00:00:00 2001
From: Michael Adler <therisen06@gmail.com>
Date: Fri, 7 May 2021 08:24:40 +0200
Subject: [PATCH 2/2] Configurable config file name

Signed-off-by: Michael Adler <therisen06@gmail.com>
---
 afew/Settings.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/afew/Settings.py b/afew/Settings.py
index 44d6f3c..a190168 100644
--- a/afew/Settings.py
+++ b/afew/Settings.py
@@ -19,7 +19,7 @@ settings = ConfigParser()
 settings.optionxform = str
 
 settings.readfp(open(os.path.join(os.path.dirname(__file__), 'defaults', 'afew.config')))
-settings.read(os.path.join(user_config_dir, 'config'))
+settings.read(os.path.join(user_config_dir, os.environ.get('AFEW_CONFIG', 'config')))
 
 # All the values for keys listed here are interpreted as ;-delimited lists
 value_is_a_list = ['tags', 'tags_blacklist']
-- 
2.31.1

