--- old/3rd/bee.lua/bee/fsevent/fsevent_linux.cpp	2021-10-11 10:00:25.585424819 +0200
+++ new/3rd/bee.lua/bee/fsevent/fsevent_linux.cpp	2021-10-11 10:00:32.039395195 +0200
@@ -23,7 +23,7 @@
             m_fd_path.emplace(std::make_pair(desc, path));
             m_path_fd.emplace(std::make_pair(path, desc));
         }
-        for (auto& p : fs::directory_iterator(path)) {
+        for (auto& p : fs::directory_iterator(path, fs::directory_options::skip_permission_denied)) {
             if (fs::is_directory(p)) {
                 add_dir(p);
             }
@@ -33,7 +33,7 @@
         int desc = m_path_fd[path];
         inotify_rm_watch(m_inotify_fd, desc);
         del_dir(desc);
-        for (auto& p : fs::directory_iterator(path)) {
+        for (auto& p : fs::directory_iterator(path, fs::directory_options::skip_permission_denied)) {
             if (fs::is_directory(p)) {
                 del_dir(p);
             }
